test/fgpu/test_engine.py

```diff
@@ -95,28 +99,124 @@ def __call__(self, t: ArrayLike) -> np.ndarray:
         return self.magnitude * np.cos(np.pi * self.frac_channel * (t - self.delay) + self.phase)
 
 
+def frac_channel(output: Output, channel: float) -> float:
+    """Convert a channel number to a `frac_channel` parameter for :class:`CW`."""
+    if isinstance(output, NarrowbandOutput):
+        # Convert centre frequency to a frac_channel
+        offset = output.centre_frequency / (ADC_SAMPLE_RATE / 2)
+        return (channel - output.channels // 2) / (output.channels * output.decimation) + offset
+    else:
+        return channel / output.channels
```

Comment:
More bike-shedding, but I wonder if this doesn't imply that the docstring of `frac_channel` shouldn't maybe be updated to something like "Frequency, as a fraction of the overall digitised bandwidth." - it's not simply a channel number divided by the number of channels, because of this offset that you're working in.

Reply:
Good call. I've updated the docstring - see what you think.